### ISSUE TITLE
glib version

### ISSUE THREAD

**Issue by bengfarrell** [ircam-ismm/node-web-audio-api#75]
Hey, forgive the lack of Linux/C expertise to know if this is even your issue.
I'm trying to get an effects processor going in Node. This project seems like probably the most promising avenue in terms of implementing the Web Audio API.

I've got this running great on my Mac, but I intended to get this working on my now outdated Jetson Nano dev kit that I bought several years back.

It ships with Ubuntu 18, and I managed to get Ubuntu 20 on there with an unofficial image.
Anyway, everything installs fine, but when I run it I get an error complaining that GLIB 2.35 can't be found.

I understand that to get that specific version, I'd need to go Ubuntu 22, or build from source and introduce some potential instability on my system.

But I guess my question is if there's a hard requirement on that version of GLIB and Ubuntu 20 won't work? Or is there just a bug and older versions will work and I can do something easy to have it pull in my current OS's version?

thanks! This is an amazing project BTW. If I can't get it working I'm just gonna get an RPI instead

**Comment by bengfarrell** [comment]
Thanks! Yeah, I think I'll try to build GLIB from source then and see what happens. At this point, if I get pushed to an RPI 5, it has better specs anyway, so I can't complain

**Comment by b-ma** [comment]
Maybe more simple would be to try to build this lib first, cf. https://github.com/ircam-ismm/node-web-audio-api?tab=readme-ov-file#manual-build

**Comment by bengfarrell** [comment]
I think you mentioned building from source before, and I mistook that as I was laser focused on building another GLIB version from source. You're right! It worked, but now I've run into other issues. The process is warning of some buffer underruns and it only keeps up so well. You do mention this in your README, so now I'm just having issues getting Jack working. I think I'm just gonna bail on the Jetson and go RPI 5. But I did manage to get my live vocal delay effect working using the Web Audio API in node, so I'm psyched. Thanks again for the great project!

**Comment by b-ma** [comment]
Cool! 

Indeed for this kind of application, ALSA wont do the job. You can have a look at [pipewire](https://pipewire.org/) too , we tested it recently as an alternative to Jack and it seems to work quite well. I don't really realize what it means to make it work though, didn't do that myself.

For RPis, maybe you would be interested with that then https://github.com/ircam-ismm/dotpi-install, that's still a work in progress and we might need some step by step guide, but it works quite well already

**Comment by bengfarrell** [comment]
Thanks again! And tanks for the dotpi install suggestion. Not sure if I'm out of the loop, but I'm not familiar with dotpi. I can't tell what it is from your repo, and I'm not coming up with anything on google. It sounds like it might be an alternative OS, or maybe stripped down mod of Raspbian geared towards audio maybe?

**Comment by b-ma** [comment]
> Not sure if I'm out of the loop, but I'm not familiar with dotpi.

Yup that's very normal, the library is still in development. We indeed need to improve the doc but in short this is mainly a set shell scripts that allows you to prepare and configure a raw Raspberry Pi OS for node / audio applications in few minutes. It also installs a client for controlling fleets of devices in distributed settings, but this is rather a edge use case and we could make that optional if needed I imagine.

Let me know if you think it might work for you, that could be an occasion for us to improve the doc
